bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/types/BArrayType.java

```diff
@@ -128,8 +128,32 @@ public boolean equals(Object obj) {
 
     @Override
     public String toString() {
-        StringBuilder sb = new StringBuilder(elementType.toString());
-        return size != -1 ? sb.append("[").append(size).append("]").toString() : sb.append("[]").toString();
+        StringBuilder sb = new StringBuilder();
+        sb.append(getTypeName());
+        sb.append(getSizeString());
+        BType element = elementType;
+        while (element instanceof BArrayType) {
+            sb.append(((BArrayType) element).getSizeString());
+            if (!(((BArrayType) element).elementType instanceof BArrayType)) {
```

Comment:
I don't think we need this line, since already checked in the `while`.

Reply:
No. we are here checking the element type of the element which is previously checked in while. Those are different checks.

Reply:
it will be checked in the beginning of the next loop, won't it?

Reply:
Yes. Removed them in [this](https://github.com/ballerina-platform/ballerina-lang/pull/22708/commits/bb556716f940f13eaf2400c2efa2442a5eb18dda).